src/Core/Cart/Calculator.php

```diff
@@ -148,12 +148,9 @@ public function getTotal($ignoreProcessedFlag = false)
             throw new \Exception('Cart must be processed before getting its total');
         }
 
-        $amount = new AmountImmutable();
-        foreach ($this->cartRows as $cartRow) {
-            $rowPrice = $cartRow->getFinalTotalPrice();
-            $amount = $amount->add($rowPrice);
-        }
-        $shippingFees = $this->fees->getFinalShippingFees();
+        $amount = $this->getRowTotalWithoutDiscount();
+        $amount = $amount->sub($this->getDiscountTotal());
+        $shippingFees = $this->fees->getInitialShippingFees();
```

Comment:
Could you please explain the reason behind using `getInitialShippingFees()` instead of `getFinalShippingFees`? using  `getFinalShippingFees` makes avoiding [#9586 ](https://github.com/PrestaShop/PrestaShop/issues/9586)(at least).